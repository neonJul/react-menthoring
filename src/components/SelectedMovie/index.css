.selected-movie {    
    /* background-image: url('../../../static/img/headerBackground.jpg'); */
    /* background-position: center; */
    background-color: black;
    padding: 30px 100px;  
    color: white;   
}

.selected-movie-search {
    background-color: white;
    color: rgb(177, 13, 95);
    float: right;
    padding: 5px 10px; 
}

.movie-description {
    display: flex;
    margin-top: 50px;
}

.movie-description-text {
    margin-left: 40px;
}

.part1 {
    display: flex;
    align-items: center;
}

h3 {
    color: rgb(177, 13, 95);
}

.part1 span {
    margin-left: 10px;
    border: 1px solid white;
    border-radius: 50%;
    padding: 15px;
}

.part2 {
    margin-bottom: 25px;
}

.part2 span {
    margin-right: 10px;
}