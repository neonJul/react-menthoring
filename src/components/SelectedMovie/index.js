import React from 'react';
import './index.css';
import Button from '../Shared/Button';
import { connect } from 'react-redux';
import {fetchMovieAsync, fetchMoviesAsync} from '../../actions/index';
import { NavLink } from 'react-router-dom';
import MovieList from "../MovieList";

class SelectedMovie extends React.Component {

    componentDidMount() {
        this.props.fetchMovieAsync(this.props.match.params.id);
    }

    componentDidUpdate(oldProps) {
        if (this.props.match.params.id !== oldProps.match.params.id) {
            this.props.fetchMovieAsync(this.props.match.params.id);
        }
    }

    render() {
        return(
            <>
                <div className="selected-movie">
                    <NavLink className="movie-item" to="/">
                        <Button name="Search" className="selected-movie-search"/>
                    </NavLink>

                    <div className="movie-description">
                        <img src={this.props.movie.poster_path}/>
                        <div className="movie-description-text">
                            <div className="part1">
                                <h3>{this.props.movie.title}</h3>
                                <span>{this.props.movie.vote_average}</span>
                            </div>
                            <div className="part2">
                                <span>{this.props.movie.release_date}</span>
                            </div>
                            <p>{this.props.movie.overview}</p>
                        </div>
                    </div>
                </div>
                <MovieList/>
            </>
        );
    }
}

const mapStateToProps = state => {
    return {
        movie: state.movie,
        movies: state.movies,
        searchParams: state.searchParams
    };
  };

const mapDispatchToProps = {
    fetchMovieAsync,
    fetchMoviesAsync
  };

export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(SelectedMovie);