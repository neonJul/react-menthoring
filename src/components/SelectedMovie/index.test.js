import React from 'react';
import renderer from 'react-test-renderer';

import SelectedMovie from './index';

it('renders correctly', () => {
  const tree = renderer.create(<SelectedMovie />).toJSON();
  expect(tree).toMatchSnapshot();
});