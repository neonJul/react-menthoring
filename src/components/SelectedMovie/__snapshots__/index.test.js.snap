// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="selected-movie"
>
  <button
    className="selected-movie-search button"
  >
    Search
  </button>
  <div
    className="movie-description"
  >
    <img
      src="test-file-stub"
    />
    <div
      className="movie-description-text"
    >
      <div
        className="part1"
      >
        <h3>
          Movie Name
        </h3>
        <span>
          4.3
        </span>
      </div>
      <div
        className="part2"
      >
        <span>
          year
        </span>
        <span>
          duration
        </span>
      </div>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      </p>
    </div>
  </div>
</div>
`;
